### DISCUSSION TITLE
SplitView for Sections with a Tree Panel

### DISCUSSION THREAD

**Discussion by idseefeld** [umbraco/Umbraco.UI#428] [General]
Hi,
I have noticed there is currently no _SplitView_ component which might be useful for tree / content sections.
Are there any plans for this?

With _SplitView_ I mean a component with two or more resizable areas like in the current backoffice for trees and content panels.
An example would be: <https://www.telerik.com/kendo-angular-ui/components/layout/splitter/>

Dirk

**Comment by idseefeld** [comment]
Another simple example: <https://opensource.adobe.com/spectrum-web-components/components/split-view/>

**Comment by bjarnef** [comment]
I think something like this would be useful.
https://shoelace.style/components/split-panel

It seems the backoffice implementation also not yet allow to adjust size of tree sidebar.

@iOvergaard I like the component above which is more flexible that the current split-panel in backoffice, e.g. to set `orientation` (vertical/horizontal), `min` and `max` (as the current backoffice UI breaks a bit when splitter is drag all to the left or right).

**Reply by iOvergaard** [reply]
That is a very cool component. We currently have a custom implementation in the backoffice. I think we could aim for something like this in the future replacing the implementation in the backoffice.

**Reply by bjarnef** [reply]
@iOvergaard I have started a draft PR here https://github.com/umbraco/Umbraco.UI/pull/594 .. but I not sure why the `<slot>` doesn't render the content, e.g. start slot could contain the tree and end area the main area.

I like the idea having "start" and "end" slots as it can be vertical (so left/right would be confusing).

Maybe we also need to consider direction (LTR or RTL) in general.
Not sure if the current backoffice handle this, but I guess in Arabic the tree would be on the right and the main area on the left.
https://www.andiamo.co.uk/resources/right-to-left-languages/
